realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -1226,6 +1226,134 @@ static void TableQuery_StringPredicate(JNIEnv* env, jlong nativeQueryPtr, jlongA
+        jsize arr_len = col_key_arr.size();
+        LinkChain linkChain = getTableForLinkQuery(nativeQueryPtr, table_arr, col_key_arr);
+
+        UUID uuid = UUID(StringData(data).data());
+        if (arr_len == 1) {
+            if (!TYPE_VALID(env, Q(nativeQueryPtr)->get_table(), col_key_arr[0], type_UUID)) {
+                return;
+            }
+
+            switch (predicate) {
+                case UUIDEqual:
+                    Q(nativeQueryPtr)->equal(ColKey(col_key_arr[0]), uuid);
+                    break;
+                case UUIDNotEqual:
+                    Q(nativeQueryPtr)->not_equal(ColKey(col_key_arr[0]), uuid);
+                    break;
+                case UUIDLess:
+                    Q(nativeQueryPtr)->less(ColKey(col_key_arr[0]), uuid);
+                    break;
+                case UUIDLessEqual:
+                    Q(nativeQueryPtr)->less_equal(ColKey(col_key_arr[0]), uuid);
+                    break;
+                case UUIDGreater:
+                    Q(nativeQueryPtr)->greater(ColKey(col_key_arr[0]), uuid);
+                    break;
+                case UUIDGreaterEqual:
+                    Q(nativeQueryPtr)->greater_equal(ColKey(col_key_arr[0]), uuid);
+                    break;
+            }
+        }
+        else {
+            LinkChain linkChain = getTableForLinkQuery(nativeQueryPtr, table_arr, col_key_arr);
+            switch (predicate) {
+                case UUIDEqual:
+                    Q(nativeQueryPtr)
+                            ->and_query(linkChain.column<UUID>(ColKey(col_key_arr[arr_len - 1])) ==
+                                        uuid);
+                    break;
+                case UUIDNotEqual:
+                    Q(nativeQueryPtr)
+                            ->and_query(linkChain.column<UUID>(ColKey(col_key_arr[arr_len - 1])) !=
+                                        uuid);
+                    break;
+                case UUIDLess:
+                    Q(nativeQueryPtr)
+                            ->and_query(numeric_link_less<UUID, UUID, UUID>(linkChain, col_key_arr[arr_len - 1], uuid));
+                    break;
+                case UUIDLessEqual:
+                    Q(nativeQueryPtr)
+                            ->and_query(numeric_link_lessequal<UUID, UUID, UUID>(linkChain, col_key_arr[arr_len - 1], uuid));
+                    break;
+                case UUIDGreater:
+                    Q(nativeQueryPtr)
+                            ->and_query(numeric_link_greater<UUID, UUID, UUID>(linkChain, col_key_arr[arr_len - 1], uuid));
+                    break;
+                case UUIDGreaterEqual:
+                    Q(nativeQueryPtr)
+                            ->and_query(numeric_link_greaterequal<UUID, UUID, UUID>(linkChain, col_key_arr[arr_len - 1], uuid));
+                    break;
+            }
+        }
+    }
+    CATCH_STD()
+}
+
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeEqualUUID(JNIEnv* env, jobject,
```

Comment:
Is there a reason to have all these below functions? Why not call TableQuery_UUIDPredicate with the additional enum?

Reply:
On this PR I was following along with the other data types, but yes we could have an additional enum and reduce the number of functions.